qiita_pet/handlers/analysis_handlers/tests/test_sharing_handlers.py

```diff
@@ -0,0 +1,59 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import main
+from json import loads
+
+from qiita_db.analysis import Analysis
+from qiita_db.user import User
+from qiita_pet.test.tornado_test_base import TestHandlerBase
+
+
+class TestShareStudyAjax(TestHandlerBase):
+
+    def test_get(self):
+        a = Analysis(1)
+        u = User('shared@foo.bar')
+        self.assertEqual(a.shared_with, [u])
+
+        # deselecting
+        args = {'deselected': u.id, 'id': a.id}
+        response = self.get('/analysis/sharing/', args)
+        self.assertEqual(response.code, 200)
+        exp = {'users': [], 'links': ''}
```

Comment:
When analyses are unshared the output data should be either empty or include a "completed" message. We were doing this for the portion of the REST API that @wasade and I worked on. Seemed clean and you don't really need an empty list of `users` or `links`. What do you think?